var/spack/repos/builtin/packages/go-bootstrap/package.py

```diff
@@ -18,50 +19,48 @@ class GoBootstrap(Package):
 
     extendable = True
 
-    # NOTE: Go@1.4.x is the only supported bootstrapping compiler because all
-    # later versions require a Go compiler to build.
-    # See: https://golang.org/doc/install/source#go14 and
-    # https://github.com/golang/go/issues/17545 and
-    # https://github.com/golang/go/issues/16352
-    version(
-        "1.4-bootstrap-20171003",
-        sha256="f4ff5b5eb3a3cae1c993723f3eab519c5bae18866b5e5f96fe1102f0cb5c3e52",
-        url="https://dl.google.com/go/go1.4-bootstrap-20171003.tar.gz",
-    )
-    version(
-        "1.4-bootstrap-20170531",
-        sha256="49f806f66762077861b7de7081f586995940772d29d4c45068c134441a743fa2",
-        url="https://storage.googleapis.com/golang/go1.4-bootstrap-20170531.tar.gz",
-    )
-    version(
-        "1.4-bootstrap-20161024",
-        sha256="398c70d9d10541ba9352974cc585c43220b6d8dbcd804ba2c9bd2fbf35fab286",
-        url="https://storage.googleapis.com/golang/go1.4-bootstrap-20161024.tar.gz",
-    )
-
-    provides("golang@:1.4-bootstrap-20171003")
+    maintainers("alecbcs")
 
     depends_on("git", type=("build", "link", "run"))
 
-    conflicts(
-        "os=monterey",
-        msg="go-bootstrap won't build on MacOS Monterey: "
-        "try `brew install go` and `spack external find go`",
-    )
-    conflicts("target=aarch64:", msg="Go bootstrap doesn't support aarch64 architectures")
-
-    # This virtual package allows a fallback to gccgo for aarch64,
-    # where go-bootstrap cannot be built(aarch64 was added with Go 1.5)
-    provides("go-external-or-gccgo-bootstrap")
-
-    # Support for aarch64 was added in Go 1.5, use an external package or gccgo instead:
-    conflicts("@:1.4", when="target=aarch64:")
-
     executables = ["^go$"]
 
+    # List binary go releases for multiple operating systems and architectures.
+    # These binary versions are not intended to stay up-to-date. Instead we
+    # should update these binary releases on a yearly schedule as
+    # bootstrapping requirements are modified by new releases of go.
+    go_releases = {
+        "1.17.13": {
+            "darwin": {
+                "amd64": "c101beaa232e0f448fab692dc036cd6b4677091ff89c4889cc8754b1b29c6608",
+                "arm64": "e4ccc9c082d91eaa0b866078b591fc97d24b91495f12deb3dd2d8eda4e55a6ea",
+            },
+            "linux": {
+                "amd64": "4cdd2bc664724dc7db94ad51b503512c5ae7220951cac568120f64f8e94399fc",
+                "arm64": "914daad3f011cc2014dea799bb7490442677e4ad6de0b2ac3ded6cee7e3f493d",
+                "ppc64le": "bd0763fb130f8412672ffe1e4a8e65888ebe2419e5caa9a67ac21e8c298aa254",
+            },
+        }
+    }
+
+    # Normalize architectures returned by platform to those used by the
+    # Go project.
+    go_targets = {"aarch64": "arm64", "arm64": "arm64", "ppc64le": "ppc64le", "x86_64": "amd64"}
+
+    # determine system os and architecture/target
+    os = platform.system().lower()
+    target = go_targets[platform.machine().lower()]
```

Comment:
Concerning the Windows failure, does `platform.machine()` return `amd64`?

Reply:
I don't believe so, on all of the intel/amd linux boxes I've tested it on its returned `x86_64`.

Reply:
Perhaps Windows is different however, I'll test it out.

Reply:
Sweet looks like Windows does return `amd64` instead of `x86_64` like MacOS and Linux.